erdpy/simulation.py

```diff
@@ -1,9 +1,17 @@
 from collections import OrderedDict
-from typing import Any, Dict
-from erdpy.interfaces import IElrondProxy, ISimulateCostResponse, ISimulateResponse, ITransaction
+from typing import Any, Dict, Protocol
+from erdpy.interfaces import ISimulateCostResponse, ISimulateResponse, ITransaction
 from erdpy.utils import ISerializable
 
 
+class INetworkProvider(Protocol):
+    def simulate_transaction(self, transaction: Dict[str, Any]) -> ISimulateResponse:
+        ...
+    
+    def simulate_transaction_cost(self, transaction: Dict[str, Any]) -> ISimulateCostResponse:
```

Comment:
Maybe drop this?

Reply:
Yes, this will be dropped. If i remember correctly we've discussed to remove `simulate_transaction_cost` in `v4.0.0`, but I was thinking to do it in a separate PR. What do you think?

Reply:
OK, can be done in a separate PR :+1: